### ISSUE TITLE
bug: database memory surges when running a lot of small batch queries

### ISSUE THREAD

**Issue by wangdexinhp** [risingwavelabs/risingwave#5724]
### Describe the bug

When starting a cluster service through docker-compose, create a simple table (only four very simple fields) in the database, and then write data, and find that the memory grows rapidly. When 300,000 data are written, the memory has grown 10G or so. And there is no response to the query on the table at this time.(select count(*) from t).

create table t (id int,v1 int,v2 int,s1 varchar);

```python
sql = "insert into t1 values({},{},{},'test_value');"
for i in range(1,1000000):
    sql = sql.format(i,i+1,i+2)
    cursor = conn.cursor()
    cursor.execute(sql)
```

### To Reproduce

1. docker compose -f docker-compose.yml up -d
2. psql -h localhost -p 4566 -d dev -U root
3. create table t (id int,v1 int,v2 int,s1 varchar);
4. insert into t values({},{},{},'test_value');

this is docker-componse.yml

```yml
version: "3"
services:
  compactor-0:
    image: "ghcr.io/risingwavelabs/risingwave:v0.1.12"
    command:
      - compactor-node
      - "--host"
      - "0.0.0.0:6660"
      - "--state-store"
      - "hummock+minio://hummockadmin:hummockadmin@minio-0:9308/hummock001"
      - "--meta-address"
      - "http://meta-node-0:5690"
    expose:
      - "6660"
    ports: []
    depends_on:
      - meta-node-0
      - minio-0
    environment:
      RUST_BACKTRACE: "1"
    container_name: compactor-0
    
  compute-node-0:
    image: "ghcr.io/risingwavelabs/risingwave:v0.1.12"
    command:
      - compute-node
      - "--host"
      - "0.0.0.0:5688"
      - "--client-address"
      - "compute-node-0:5688"
      - "--state-store"
      - "hummock+minio://hummockadmin:hummockadmin@minio-0:9308/hummock001"
      - "--meta-address"
      - "http://meta-node-0:5690"
    expose:
      - "5688"
    ports: []
    depends_on:
      - meta-node-0
      - minio-0
    environment:
      RUST_BACKTRACE: "1"
    container_name: compute-node-0
  etcd-0:
    image: "quay.io/coreos/etcd:latest"
    command:
      - /usr/local/bin/etcd
      - "--listen-client-urls"
      - "http://0.0.0.0:2388"
      - "--advertise-client-urls"
      - "http://etcd-0:2388"
      - "--listen-peer-urls"
      - "http://0.0.0.0:2389"
      - "--initial-advertise-peer-urls"
      - "http://etcd-0:2389"
      - "--listen-metrics-urls"
      - "http://0.0.0.0:2379"
      - "--name"
      - risedev-meta
      - "--max-txn-ops"
      - "999999"
      - "--auto-compaction-mode"
      - periodic
      - "--auto-compaction-retention"
      - 1m
      - "--snapshot-count"
      - "10000"
    expose:
      - "2388"
    ports:
      - "2388:2388"
      - "2389:2389"
    depends_on: []
    volumes:
      - "/Users/wangdexin/mnt/etcd/etcd-0:/etcd-data"
    environment: {}
    container_name: etcd-0
  frontend-node-0:
    image: "ghcr.io/risingwavelabs/risingwave:v0.1.12"
    command:
      - frontend-node
      - "--host"
      - "0.0.0.0:4566"
      - "--meta-addr"
      - "http://meta-node-0:5690"
    expose:
      - "4566"
    ports:
      - "4566:4566"
    depends_on:
      - meta-node-0
    environment:
      RUST_BACKTRACE: "1"
    container_name: frontend-node-0
    
  
  meta-node-0:
    image: "ghcr.io/risingwavelabs/risingwave:v0.1.12"
    command:
      - meta-node
      - "--listen-addr"
      - "0.0.0.0:5690"
      - "--host"
      - meta-node-0
      - "--backend"
      - etcd
      - "--etcd-endpoints"
      - "etcd-0:2388"
    expose:
      - "5690"
      - "1250"
      - "5691"
    ports: []
    depends_on: []
    environment:
      RUST_BACKTRACE: "1"
    container_name: meta-node-0
  
  minio-0:
    image: "quay.io/minio/minio:latest"
    command:
      - server
      - "--address"
      - "0.0.0.0:9308"
      - "--console-address"
      - "0.0.0.0:9400"
      - /data
    expose:
      - "9308"
      - "9400"
    ports:
      - "9308:9308"
      - "9400:9400"
    depends_on: []
    volumes:
      - "/Users/wangdexin/mnt/minio/minio-0:/data"
    entrypoint: "\n/bin/sh -c '\nset -e\nmkdir -p \"/data/hummock001\"\n/usr/bin/docker-entrypoint.sh \"$$0\" \"$$@\"\n'"
    environment:
      MINIO_CI_CD: "1"
      MINIO_ROOT_PASSWORD: hummockadmin
      MINIO_ROOT_USER: hummockadmin
    container_name: minio-0

  zookeeper:
    image: confluentinc/cp-zookeeper:7.0.1
    environment:
      ZOOKEEPER_CLIENT_PORT: 2181
      ZOOKEEPER_TICK_TIME: 2000
      KAFKA_OPTS: "-Dzookeeper.4lw.commands.whitelist=ruok"
    expose:
      - "2181"
    ports:
      - "2181:2181"
    healthcheck:
      {
        test: echo ruok | nc localhost 2181 || exit -1,
        interval: 1s,
        timeout: 5s,
        retries: 3,
        start_period: 120s
      }

  kafka:
    image: wurstmeister/kafka
    environment:
      #KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
      #KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://127.0.0.1:9092
      #KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
      #KAFKA_COMPRESSION_TYPE: lz4
      #KAFKA_LOG_RETENTION_MS: 31104000000
      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: INSIDE:PLAINTEXT,OUTSIDE:PLAINTEXT
      KAFKA_LISTENERS: INSIDE://:9092,OUTSIDE://127.0.0.1:9091
      KAFKA_ADVERTISED_LISTENERS: INSIDE://:9092,OUTSIDE://127.0.0.1:9091
      KAFKA_INTER_BROKER_LISTENER_NAME: INSIDE
    depends_on:
      zookeeper: { condition: service_healthy }
    expose:
      - "9092"
    ports:
      - "9091:9091"
    healthcheck:
      {
        test: nc -z localhost 9092,
        interval: 5s,
        start_period: 120s
      }



volumes:
  etcd-0:
    external: false
  minio-0:
    external: false

name: risingwave-compose
```

### Expected behavior

The memory grows quickly, and when more than 300,000 data is inserted, the query does not respond.
select count(*) from t;

### Additional context

This is the result whether I create a materialized view based on the table or not.

> 

Oh, I glued it with indentation, and it was gone when the submission was completed. You can click edit and you can copy the indented yml.

**Comment by Gun9niR** [comment]
Which component has reached 10G memory usage? Compute node or compactor?

**Comment by Gun9niR** [comment]
BTW, could you put the yml in a code snippet so that we can copy & paste it more conveniently?

**Comment by wangdexinhp** [comment]
> 哪个组件的内存使用量达到了 10G？计算节点还是压缩器？

The whole adds up to about 10G

**Comment by wangdexinhp** [comment]
> Which component has reached 10G memory usage? Compute node or compactor?

The whole adds up to about 10G，

**Comment by wangdexinhp** [comment]
> BTW, could you put the yml in a code snippet so that we can copy & paste it more conveniently?

yes, I have written the yaml file, in the middle of `Describe the bug`

**Comment by Gun9niR** [comment]
> > BTW, could you put the yml in a code snippet so that we can copy & paste it more conveniently?
> 
> yes, I have written the yml file, in the middle of `Describe the bug`

I mean there's no indentation so whoever needs to reuse it needs to format it themselves🤣.

**Comment by wangdexinhp** [comment]
version: "3"
services:
  compactor-0:
    image: "ghcr.io/singularity-data/risingwave:latest"
    command:
      - compactor-node
      - "--host"
      - "0.0.0.0:6660"
      - "--prometheus-listener-addr"
      - "0.0.0.0:1260"
      - "--metrics-level"
      - "1"
      - "--state-store"
      - "hummock+minio://hummockadmin:hummockadmin@minio-0:9301/hummock001"
      - "--meta-address"
      - "http://meta-node-0:5690"
    expose:
      - "6660"
      - "1260"
    ports: []
    depends_on:
      - meta-node-0
      - minio-0
    volumes: []
    environment: {}
    container_name: compactor-0

  compute-node-0:
    image: "ghcr.io/singularity-data/risingwave:latest"
    command:
      - compute-node
      - "--host"
      - "0.0.0.0:5688"
      - "--prometheus-listener-addr"
      - "0.0.0.0:1222"
      - "--client-address"
      - "compute-node-0:5688"
      - "--metrics-level"
      - "1"
      - "--state-store"
      - "hummock+minio://hummockadmin:hummockadmin@minio-0:9301/hummock001"
      - "--meta-address"
      - "http://meta-node-0:5690"
    expose:
      - "5688"
      - "1222"
    ports: []
    depends_on:
      - meta-node-0
      - minio-0
    volumes: []
    environment: {}
    container_name: compute-node-0

  frontend-node-0:
    image: "ghcr.io/singularity-data/risingwave:latest"
    command:
      - frontend-node
      - "--host"
      - "0.0.0.0:4566"
      - "--meta-addr"
      - "http://meta-node-0:5690"
    expose:
      - "4566"
    ports:
      - "9876:4566"
    depends_on:
      - meta-node-0
    volumes: []
    environment: {}
    container_name: frontend-node-0

  meta-node-0:
    image: "ghcr.io/singularity-data/risingwave:latest"
    command:
      - meta-node
      - "--host"
      - "0.0.0.0:5690"
      - "--dashboard-host"
      - "0.0.0.0:5691"
      - "--prometheus-host"
      - "0.0.0.0:1250"
      - "--backend"
      - etcd
      - "--etcd-endpoints"
      - etcd:2379
      - --dashboard-ui-path
      - /meta-ui
    expose:
      - "5690"
      - "1250"
      - "5691"
    ports: []
    depends_on: []
    volumes: []
    environment: {}
    container_name: meta-node-0

  etcd:
    image: quay.io/coreos/etcd:latest
    hostname: etcd
    environment:
      ETCD_NAME: node1
      ETCD_INITIAL_ADVERTISE_PEER_URLS: http://etcd:2380
      ETCD_LISTEN_PEER_URLS: http://0.0.0.0:2380
      ETCD_ADVERTISE_CLIENT_URLS: http://etcd:2379
      ETCD_LISTEN_CLIENT_URLS: http://0.0.0.0:2379
      ETCD_DATA_DIR: /etcd-data
      ETCD_AUTO_COMPACTION_MODE: revision
      ETCD_AUTO_COMPACTION_RETENTION: 100
    volumes:
      - "etcd-data:/etcd-data"

  etcd-compact:
    image: quay.io/coreos/etcd:v3.5.4
    entrypoint: |
      sh -c " 
        echo '*/30 * * * * ETCDCTL_API=3 etcdctl --endpoints http://etcd:2379 defrag > /proc/1/fd/1 2>/proc/1/fd/2' > cron.conf 
        crontab cron.conf 
        crontab -l
        crond -f 
      "

  minio-0:
    image: "quay.io/minio/minio:latest"
    command:
      - server
      - "--address"
      - "0.0.0.0:9301"
      - "--console-address"
      - "0.0.0.0:9400"
      - /data
    expose:
      - "9301"
      - "9400"
    ports:
      - "9301:9301"
      - "9400:9400"
    depends_on: []
    volumes:
      - "minio-0:/data"
    entrypoint: "\n/bin/sh -c '\nset -e\nmkdir -p \"/data/hummock001\"\n/usr/bin/docker-entrypoint.sh \"$$0\" \"$$@\"\n'"
    environment:
      MINIO_PROMETHEUS_URL: "http://prometheus-0:9500"
      MINIO_PROMETHEUS_AUTH_TYPE: public
      MINIO_ROOT_PASSWORD: hummockadmin
      MINIO_ROOT_USER: hummockadmin
    container_name: minio-0

  prometheus-0:
    image: "prom/prometheus:latest"
    command:
      - "--config.file=/etc/prometheus/prometheus.yml"
      - "--storage.tsdb.path=/prometheus"
      - "--web.console.libraries=/usr/share/prometheus/console_libraries"
      - "--web.console.templates=/usr/share/prometheus/consoles"
      - "--web.listen-address=0.0.0.0:9500"
    expose:
      - "9500"
    ports:
      - "9501:9500"
    depends_on: []
    volumes:
      - "prometheus-0:/prometheus"
    entrypoint: "\n/bin/sh -c '\nset -e\necho \"$$PROMETHEUS_CONFIG\" > /etc/prometheus/prometheus.yml\n/bin/prometheus \"$$0\" \"$$@\"\n'"
    environment:
      PROMETHEUS_CONFIG: "# --- THIS FILE IS AUTO GENERATED BY RISEDEV ---\nglobal:\n  scrape_interval: 1s\n  evaluation_interval: 5s\n\nscrape_configs:\n  - job_name: \"prometheus\"\n    static_configs:\n      - targets: [\"prometheus-0:9500\"]\n\n  - job_name: compute-job\n    static_configs:\n      - targets: [\"compute-node-0:1222\"]\n\n  - job_name: meta-job\n    static_configs:\n      - targets: [\"meta-node-0:1250\"]\n  \n  - job_name: minio-job\n    metrics_path: /minio/v2/metrics/cluster\n    static_configs:\n    - targets: [\"minio-0:9301\"]\n\n  - job_name: compactor-job\n    static_configs:\n      - targets: [\"compactor-0:1260\"]\n"
    container_name: prometheus-0
    
  zookeeper:
    image: confluentinc/cp-zookeeper:7.1.0
    hostname: zookeeper
    container_name: zookeeper
    ports:
      - "4568:2181"
    environment:
      ZOOKEEPER_CLIENT_PORT: 2181
      ZOOKEEPER_TICK_TIME: 2000

  broker:
    image: confluentinc/cp-server:7.1.0
    hostname: broker
    container_name: broker
    depends_on:
      - zookeeper
    ports:
      - "4567:9092"
      - "9101:9101"
    environment:
      KAFKA_BROKER_ID: 1
      KAFKA_ZOOKEEPER_CONNECT: 'zookeeper:2181'
      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://broker:29092,PLAINTEXT_HOST://localhost:9092
      KAFKA_METRIC_REPORTERS: io.confluent.metrics.reporter.ConfluentMetricsReporter
      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
      KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS: 0
      KAFKA_CONFLUENT_LICENSE_TOPIC_REPLICATION_FACTOR: 1
      KAFKA_CONFLUENT_BALANCER_TOPIC_REPLICATION_FACTOR: 1
      KAFKA_TRANSACTION_STATE_LOG_MIN_ISR: 1
      KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR: 1
      KAFKA_JMX_PORT: 9101
      KAFKA_JMX_HOSTNAME: localhost
      KAFKA_CONFLUENT_SCHEMA_REGISTRY_URL: http://schema-registry:8081
      CONFLUENT_METRICS_REPORTER_BOOTSTRAP_SERVERS: broker:29092
      CONFLUENT_METRICS_REPORTER_TOPIC_REPLICAS: 1
      CONFLUENT_METRICS_ENABLE: 'true'
      CONFLUENT_SUPPORT_CUSTOMER_ID: 'anonymous'
    
  datagen:
    build:
      context: ../py_gen
    entrypoint: /bin/sh
    tty: true
    depends_on:
      - broker

volumes:
  minio-0:
    external: false
  prometheus-0:
    external: false
  etcd-data:
    external: false
name: risingwave-compose

**Comment by wangdexinhp** [comment]
version: "3"
services:
  compactor-0:
    image: "ghcr.io/singularity-data/risingwave:latest"
    command:
      - compactor-node
      - "--host"
      - "0.0.0.0:6660"
      - "--prometheus-listener-addr"
      - "0.0.0.0:1260"
      - "--metrics-level"
      - "1"
      - "--state-store"
      - "hummock+minio://hummockadmin:hummockadmin@minio-0:9301/hummock001"
      - "--meta-address"
      - "http://meta-node-0:5690"
    expose:
      - "6660"
      - "1260"
    ports: []
    depends_on:
      - meta-node-0
      - minio-0
    volumes: []
    environment: {}
    container_name: compactor-0

  compute-node-0:
    image: "ghcr.io/singularity-data/risingwave:latest"
    command:
      - compute-node
      - "--host"
      - "0.0.0.0:5688"
      - "--prometheus-listener-addr"
      - "0.0.0.0:1222"
      - "--client-address"
      - "compute-node-0:5688"
      - "--metrics-level"
      - "1"
      - "--state-store"
      - "hummock+minio://hummockadmin:hummockadmin@minio-0:9301/hummock001"
      - "--meta-address"
      - "http://meta-node-0:5690"
    expose:
      - "5688"
      - "1222"
    ports: []
    depends_on:
      - meta-node-0
      - minio-0
    volumes: []
    environment: {}
    container_name: compute-node-0

  frontend-node-0:
    image: "ghcr.io/singularity-data/risingwave:latest"
    command:
      - frontend-node
      - "--host"
      - "0.0.0.0:4566"
      - "--meta-addr"
      - "http://meta-node-0:5690"
    expose:
      - "4566"
    ports:
      - "9876:4566"
    depends_on:
      - meta-node-0
    volumes: []
    environment: {}
    container_name: frontend-node-0

  meta-node-0:
    image: "ghcr.io/singularity-data/risingwave:latest"
    command:
      - meta-node
      - "--host"
      - "0.0.0.0:5690"
      - "--dashboard-host"
      - "0.0.0.0:5691"
      - "--prometheus-host"
      - "0.0.0.0:1250"
      - "--backend"
      - etcd
      - "--etcd-endpoints"
      - etcd:2379
      - --dashboard-ui-path
      - /meta-ui
    expose:
      - "5690"
      - "1250"
      - "5691"
    ports: []
    depends_on: []
    volumes: []
    environment: {}
    container_name: meta-node-0

  etcd:
    image: quay.io/coreos/etcd:latest
    hostname: etcd
    environment:
      ETCD_NAME: node1
      ETCD_INITIAL_ADVERTISE_PEER_URLS: http://etcd:2380
      ETCD_LISTEN_PEER_URLS: http://0.0.0.0:2380
      ETCD_ADVERTISE_CLIENT_URLS: http://etcd:2379
      ETCD_LISTEN_CLIENT_URLS: http://0.0.0.0:2379
      ETCD_DATA_DIR: /etcd-data
      ETCD_AUTO_COMPACTION_MODE: revision
      ETCD_AUTO_COMPACTION_RETENTION: 100
    volumes:
      - "etcd-data:/etcd-data"

  etcd-compact:
    image: quay.io/coreos/etcd:v3.5.4
    entrypoint: |
      sh -c " 
        echo '*/30 * * * * ETCDCTL_API=3 etcdctl --endpoints http://etcd:2379 defrag > /proc/1/fd/1 2>/proc/1/fd/2' > cron.conf 
        crontab cron.conf 
        crontab -l
        crond -f 
      "

  minio-0:
    image: "quay.io/minio/minio:latest"
    command:
      - server
      - "--address"
      - "0.0.0.0:9301"
      - "--console-address"
      - "0.0.0.0:9400"
      - /data
    expose:
      - "9301"
      - "9400"
    ports:
      - "9301:9301"
      - "9400:9400"
    depends_on: []
    volumes:
      - "minio-0:/data"
    entrypoint: "\n/bin/sh -c '\nset -e\nmkdir -p \"/data/hummock001\"\n/usr/bin/docker-entrypoint.sh \"$$0\" \"$$@\"\n'"
    environment:
      MINIO_PROMETHEUS_URL: "http://prometheus-0:9500"
      MINIO_PROMETHEUS_AUTH_TYPE: public
      MINIO_ROOT_PASSWORD: hummockadmin
      MINIO_ROOT_USER: hummockadmin
    container_name: minio-0

  prometheus-0:
    image: "prom/prometheus:latest"
    command:
      - "--config.file=/etc/prometheus/prometheus.yml"
      - "--storage.tsdb.path=/prometheus"
      - "--web.console.libraries=/usr/share/prometheus/console_libraries"
      - "--web.console.templates=/usr/share/prometheus/consoles"
      - "--web.listen-address=0.0.0.0:9500"
    expose:
      - "9500"
    ports:
      - "9501:9500"
    depends_on: []
    volumes:
      - "prometheus-0:/prometheus"
    entrypoint: "\n/bin/sh -c '\nset -e\necho \"$$PROMETHEUS_CONFIG\" > /etc/prometheus/prometheus.yml\n/bin/prometheus \"$$0\" \"$$@\"\n'"
    environment:
      PROMETHEUS_CONFIG: "# --- THIS FILE IS AUTO GENERATED BY RISEDEV ---\nglobal:\n  scrape_interval: 1s\n  evaluation_interval: 5s\n\nscrape_configs:\n  - job_name: \"prometheus\"\n    static_configs:\n      - targets: [\"prometheus-0:9500\"]\n\n  - job_name: compute-job\n    static_configs:\n      - targets: [\"compute-node-0:1222\"]\n\n  - job_name: meta-job\n    static_configs:\n      - targets: [\"meta-node-0:1250\"]\n  \n  - job_name: minio-job\n    metrics_path: /minio/v2/metrics/cluster\n    static_configs:\n    - targets: [\"minio-0:9301\"]\n\n  - job_name: compactor-job\n    static_configs:\n      - targets: [\"compactor-0:1260\"]\n"
    container_name: prometheus-0
    
  zookeeper:
    image: confluentinc/cp-zookeeper:7.1.0
    hostname: zookeeper
    container_name: zookeeper
    ports:
      - "4568:2181"
    environment:
      ZOOKEEPER_CLIENT_PORT: 2181
      ZOOKEEPER_TICK_TIME: 2000

  broker:
    image: confluentinc/cp-server:7.1.0
    hostname: broker
    container_name: broker
    depends_on:
      - zookeeper
    ports:
      - "4567:9092"
      - "9101:9101"
    environment:
      KAFKA_BROKER_ID: 1
      KAFKA_ZOOKEEPER_CONNECT: 'zookeeper:2181'
      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://broker:29092,PLAINTEXT_HOST://localhost:9092
      KAFKA_METRIC_REPORTERS: io.confluent.metrics.reporter.ConfluentMetricsReporter
      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
      KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS: 0
      KAFKA_CONFLUENT_LICENSE_TOPIC_REPLICATION_FACTOR: 1
      KAFKA_CONFLUENT_BALANCER_TOPIC_REPLICATION_FACTOR: 1
      KAFKA_TRANSACTION_STATE_LOG_MIN_ISR: 1
      KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR: 1
      KAFKA_JMX_PORT: 9101
      KAFKA_JMX_HOSTNAME: localhost
      KAFKA_CONFLUENT_SCHEMA_REGISTRY_URL: http://schema-registry:8081
      CONFLUENT_METRICS_REPORTER_BOOTSTRAP_SERVERS: broker:29092
      CONFLUENT_METRICS_REPORTER_TOPIC_REPLICAS: 1
      CONFLUENT_METRICS_ENABLE: 'true'
      CONFLUENT_SUPPORT_CUSTOMER_ID: 'anonymous'
    
  datagen:
    build:
      context: ../py_gen
    entrypoint: /bin/sh
    tty: true
    depends_on:
      - broker

volumes:
  minio-0:
    external: false
  prometheus-0:
    external: false
  etcd-data:
    external: false
name: risingwave-compose

**Comment by wangdexinhp** [comment]
> > > BTW, could you put the yml in a code snippet so that we can copy & paste it more conveniently?
> > 
> > 
> > yes, I have written the yml file, in the middle of `Describe the bug`
> 
> I mean there's no indentation so whoever needs to reuse it needs to format it themselves🤣.

Oh, I glued it with indentation, and it was gone when the submission was completed. You can click edit and you can copy the indented yml.

**Comment by Gun9niR** [comment]
> > > > BTW, could you put the yml in a code snippet so that we can copy & paste it more conveniently?
> > > 
> > > 
> > > yes, I have written the yml file, in the middle of `Describe the bug`
> > 
> > 
> > I mean there's no indentation so whoever needs to reuse it needs to format it themselves🤣.
> 
> Oh, I glued it with indentation, and it was gone when the submission was completed. You can click edit and you can copy the indented yml.

I've edited the comment for you.

**Comment by wangdexinhp** [comment]
> > > > > BTW, could you put the yml in a code snippet so that we can copy & paste it more conveniently?
> > > > 
> > > > 
> > > > yes, I have written the yml file, in the middle of `Describe the bug`
> > > 
> > > 
> > > I mean there's no indentation so whoever needs to reuse it needs to format it themselves🤣.
> > 
> > 
> > Oh, I glued it with indentation, and it was gone when the submission was completed. You can click edit and you can copy the indented yml.
> 
> I've edited the comment for you.

Sorry, I don't use git very well. thanks

**Comment by Gun9niR** [comment]
Successfully reproduced with `risedev d`. Looks like prometheus is taking up a large portion of memory. 

<img width="1358" alt="image" src="https://user-images.githubusercontent.com/52783948/194746542-3c37407e-22e2-4014-b944-13b764fe6054.png">

**Comment by wangdexinhp** [comment]
> prometheus

But I didn't start the prometheus node

**Comment by wangdexinhp** [comment]
![image](https://user-images.githubusercontent.com/100552985/194747266-e6f55812-7d56-46f2-931d-537a40f18040.png)

**Comment by Gun9niR** [comment]
Seems that per-task metrics (it doesn't matter whether prometheus is enabled or not) is mainly responsible for the large memory usage on compute node.

<img width="1239" alt="image" src="https://user-images.githubusercontent.com/52783948/194747966-56d96540-4cab-4ca4-8da8-f6bd23e0467f.png">

**Comment by wangdexinhp** [comment]
Well, but this situation doesn't seem reasonable, the memory is growing too fast. Can this be optimized?

**Comment by Gun9niR** [comment]
> And the original version doesn't seem to have such an exaggerated memory growth

Which version is that?

I've yet to investigate the frontend node. I guess our per-task metric is not suitable in this scenario. @ZENOTME any ideas for improvement?

**Comment by wangdexinhp** [comment]
> > And the original version doesn't seem to have such an exaggerated memory growth
> 
> Which version is that?
> 
> I've yet to investigate the frontend node. I guess our per-task metric is not suitable in this scenario. @ZENOTME any ideas for improvement?

Seems like version 0.1.9, which I used at the time, didn't happen.

**Comment by Gun9niR** [comment]
Cool. We can compare these two versions and see what's wrong.

**Comment by BugenZhao** [comment]
cc @BowenXiao1999 Could you please help to take a look?

**Comment by BowenXiao1999** [comment]
BTW, though we can do some improvement to avoid soaring memory, it is not recommend to issue 300,000 sql, cuz there will cause 300, 000 batch task. 

Better concat these insert values into one sql.

**Comment by Gun9niR** [comment]
The memory issue on CN seems to be solved by #5770 

**v0.1.9**

<img width="688" alt="image" src="https://user-images.githubusercontent.com/52783948/195003431-bf7d292f-9bdd-4020-befa-d0a01846f334.png">

**5770**

![image](https://user-images.githubusercontent.com/52783948/195003549-80d29e5a-5be6-438d-a228-8bea7d612a30.png)

**Comment by Gun9niR** [comment]
FE side should be fixed by #5822 and #5827. Closing the issue.

**Before**

![image](https://user-images.githubusercontent.com/52783948/196089810-8f41229c-d1df-492e-8a0e-58eff55035e2.png)

**After**

![image](https://user-images.githubusercontent.com/52783948/196089847-0a65bb9f-0ba0-42be-8640-1c064239e398.png)
